native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -128,6 +176,43 @@ extension ViewController: UITableViewDelegate, UITableViewDataSource {
         return cell
     }
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 80.0;//Choose your custom row height
+        return 80.0
+    }
+}
+
+extension ViewController: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    func setupAdvancedSearchCollectionView() {
```

Comment:
Because this function is specifically to do with the fact your view controller is an AdvancedSearchDelegate it would make more sense to be in an extension which is implementing methods to do with that. I would expect this extension only to include the methods relevant to the fact the view controller is a UICollectionViewDataSource and DelegateFlowLayout. Do you see why?

Reply:
Not completely sure of where you mean this method should go... I'm setting the delegate and data source for the collection view but not the collectionViewCell, so I feel like it shouldn't go in the AdvancedSearchDelegate extension?

Reply:
Ah right okay, sorry think I misinterpreted what this function was doing when I looked at it yesterday. Given that it is actually setting it up for the collection view then this is the right place for it!